File: 1047 - Remove All Adjacent Duplicates In String.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function (s) {
    let res = [];
    for (let c of s) {
        if (res.length == 0 || res[res.length - 1] != c) {
            res.push(c);
        } else {
            res.pop();
        }
    }
    return res.join('');
};

File: 1047 - Remove All Adjacent Duplicates In String.py
from typing import Deque


class Solution:
    def removeDuplicates(self, s: str) -> str:
        q = Deque()
        q.append(s[0])
        for c in s[1:]:
            if len(q) == 0 or q[-1] != c:
                q.append(c)
            else:
                q.pop()
        res = list(q)
        res = ''.join(res)
        return res